Assets/Scripts/Robot/IRobotController.cs

```diff
@@ -39,7 +39,7 @@ public interface IRobotController
         /// <summary>
         /// Gets the partition this robot is assigned to.
         /// </summary>
-        int AssignedPartition { get; }
+        int AssignedPartition { get; set; }
```

Comment:
What is this variable used for?

Reply:
It is used to assign robots to new partitions when redistributing them

Reply:
It should probably not be on the robot controller, i would suggest moving it to the algorithm or the component or whatever.

Reply:
Why not?
I dont really see an issue with the robots knowing what partitions theyre assigned to?
Its possible to store it elsewhere, probably on the partition component, though it would probably be a bit of a pain. I also dont think we would gain much from it.

Reply:
Yeah but it seems arbitrary that a robot has an assigned partition.

Reply:
This is the simplest solution for us for now. I agree thats its arbitrary but i dont think its much of an issue. 
Ill refactor it in the future if it causes issues.